insights/combiners/sysctl_conf.py

```diff
@@ -0,0 +1,71 @@
+"""
+Sysctl configuration files
+==========================
+"""
+import re
+
+from insights.core.plugins import combiner
+from insights.parsers.sysctl import SysctlConf, SysctlDConfEtc, SysctlDConfUsr
+
+
+@combiner([SysctlConf, SysctlDConfUsr, SysctlDConfEtc])
+class SysctlConfs(dict):
+    """
+    Combiner for accessing all the sysctl configuration files in one
+    structure.
+
+    Sample input::
+
+        # sysctl.conf sample
+        #
+          kernel.domainname = example.com
+
+        ; this one has a space which will be written to the sysctl!
+          kernel.modprobe = /sbin/mod probe
+
+    Attributes:
+        search(dict): Returns a dict of any kv pairs where the
+        contains the search word.
+
+    Examples:
+        >>> type(sysctl_conf)
+        <class 'insights.combiners.sysctl_conf.SysctlConfs'>
+        >>> sysctl_conf['kernel.domainname']
+        'example.com'
+        >>> sysctl_conf['kernel.modprobe']
+        '/sbin/mod probe'
+        >>> sysctl_conf['kernel.sysrq']
+        '1'
+        >>> "vm.dirty_ratio" in sysctl_conf
+        True
+        >>> sysctl_conf.search("domainname")
+        {'kernel.domainname': 'example.com'}
+    """
+    def __init__(self, sysctl_conf, sysctl_d_confs_usr, sysctl_d_confs_etc):
+        super(SysctlConfs, self).__init__()
+        if sysctl_conf:
+            self.update(sysctl_conf)
```

Comment:
@ryan-blakley - it seems that the current order doesn't match with the sysctl-manual
```
# man sysctl.d

CONFIGURATION DIRECTORIES AND PRECEDENCE
       Configuration files are read from directories in /etc/, /run/,
       /usr/local/lib/, and /usr/lib/, in order of precedence, as listed
       in the SYNOPSIS section above. Files must have the ".conf"
       extension. Files in /etc/ override files with the same name in
       /run/, /usr/local/lib/, and /usr/lib/. Files in /run/ override
       files with the same name under /usr/.
```

And the `sysctl --system` gives the following output:
```
# sudo sysctl --system
* Applying /usr/lib/sysctl.d/10-default-yama-scope.conf ...
kernel.yama.ptrace_scope = 0
* Applying /usr/lib/sysctl.d/50-coredump.conf ...
kernel.core_pattern = |/usr/lib/systemd/systemd-coredump %P %u %g %s %t %c %h %e
kernel.core_pipe_limit = 16
* Applying /usr/lib/sysctl.d/50-default.conf ...
kernel.sysrq = 16
kernel.core_uses_pid = 1
kernel.kptr_restrict = 1
net.ipv4.conf.all.rp_filter = 1
net.ipv4.conf.all.accept_source_route = 0
net.ipv4.conf.all.promote_secondaries = 1
net.core.default_qdisc = fq_codel
fs.protected_hardlinks = 1
fs.protected_symlinks = 1
* Applying /usr/lib/sysctl.d/50-libkcapi-optmem_max.conf ...
net.core.optmem_max = 81920
* Applying /usr/lib/sysctl.d/50-pid-max.conf ...
kernel.pid_max = 4194304
* Applying /usr/lib/sysctl.d/60-libvirtd.conf ...
fs.aio-max-nr = 1048576
* Applying /usr/lib/sysctl.d/88-abc.conf ...
kernel.shmmax = 18446744073692774390
* Applying /etc/sysctl.d/99-abc.conf ...
kernel.shmmax = 18446744073692774380
* Applying /etc/sysctl.d/99-sysctl.conf ...
kernel.shmmax = 18446744073692774377
* Applying /etc/sysctl.d/99-xyz.conf ...
kernel.shmmax = 18446744073692774388
* Applying /etc/sysctl.conf ...
kernel.shmmax = 18446744073692774377
```

```
# sudo sysctl -a | grep shmmax
kernel.shmmax = 18446744073692774377
```

It seems that the correct order in the combiner should be:
```
/usr/lib/sysctl.d/*.conf  -> /etc/sysctl.d/*.conf -> /etc/sysctl.conf
```